neural_modelling/src/neuron/plasticity/stdp/synapse_dynamics_stdp_mad_impl.c

```diff
@@ -93,7 +93,8 @@ static inline final_state_t _plasticity_update_synapse(
     // Process events in post-synaptic window
     while (post_window.num_events > 0) {
         const uint32_t delayed_post_time = *post_window.next_time
-                                           + delay_dendritic;
+//                                           + delay_dendritic;
+                                           + 0;
```

Comment:
I am slightly worried about the effect of this on normal plasticity.  I can't remember the details, but is this correct?

Reply:
It depend what correct means. The default implementation on spinnaker is and has been for a while to apply a `delay_dendritic` backpropagation delay, however this behaves incorrectly for autapses. I believe Steve wanted us to resolve this somehow.

Reply:
I am looking back at the description in our paper, and trying to work out which is the thing we do.  Is the problem that if delay_dendritic is used instead of 0, STDP never has any effect on the autapse?  Or is it that the autapse is always depressed?

Additionally, if we use 0 here, what effect does this have on STDP on non-autapses?

Reply:
I think I remember now that you want autapses to be depressed, because a spike can never have caused itself.  I guess our current code disallows this so autapses always continue to exist... I will have to think if there is an easy solution to this.